tests/rust_reflection_test/src/lib.rs

```diff
@@ -0,0 +1,1324 @@
+    let verify_options = VerifierOptions {
+        max_apparent_size: 1 << 6,
+        ..Default::default()
+    };
+
+    let res = verify_with_options(&buffer, &schema, &verify_options);
+
+    assert!(res.is_err());
+    assert_eq!(format!("{:#?}", res.err().unwrap()), "ApparentSizeTooLarge");
+}
+
+fn load_file_as_buffer(path: &str) -> Vec<u8> {
+    let mut file = File::open(path).unwrap();
+    let mut data = Vec::new();
+    file.read_to_end(&mut data).unwrap();
+    data
+}
+
+fn get_schema_field<'a>(schema_buffer: &'a Vec<u8>, field_name: &'a str) -> Field<'a> {
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+    let root_table = schema.root_table().unwrap();
+    let fields = root_table.fields();
+    fields
+        .lookup_by_key(field_name, |field, key| field.key_compare_with_value(key))
+        .unwrap()
+}
+
+fn create_test_buffer() -> Vec<u8> {
+    let mut builder = FlatBufferBuilder::new();
+    let mon = {
+        let s0 = builder.create_string("test1");
+        let s1 = builder.create_string("test2");
+        let fred_name = builder.create_string("Fred");
+
+        // can't inline creation of this Vec3 because we refer to it by reference, so it must live
+        // long enough to be used by MonsterArgs.
+        let pos = my_game::example::Vec3::new(
+            1.0,
+            2.0,
+            3.0,
+            3.0,
+            my_game::example::Color::Green,
+            &my_game::example::Test::new(5i16, 6i8),
+        );
+
+        let args = my_game::example::MonsterArgs {
+            hp: 32767,
+            testf: 3.14,
+            mana: 150,
+            name: Some(builder.create_string("MyMonster")),
+            pos: Some(&pos),
+            test_type: my_game::example::Any::Monster,
+            testempty: Some(my_game::example::Stat::create(
+                &mut builder,
+                &my_game::example::StatArgs {
+                    id: Some(fred_name),
+                    ..Default::default()
+                },
+            )),
+            test: Some(
+                my_game::example::Monster::create(
+                    &mut builder,
+                    &my_game::example::MonsterArgs {
+                        name: Some(fred_name),
+                        ..Default::default()
+                    },
+                )
+                .as_union_value(),
+            ),
+            inventory: Some(builder.create_vector(&[0u8, 1, 2, 3, 4])),
+            test4: Some(builder.create_vector(&[
+                my_game::example::Test::new(10, 20),
+                my_game::example::Test::new(30, 40),
+            ])),
+            testarrayofstring: Some(builder.create_vector(&[s0, s1])),
+            ..Default::default()
+        };
+        my_game::example::Monster::create(&mut builder, &args)
+    };
+    my_game::example::finish_monster_buffer(&mut builder, mon);
+    let (flatbuf, start_loc) = builder.mut_finished_buffer();
+    flatbuf[start_loc..].to_vec()
+}
```

Comment:
Whoa, that's a lot of tests. Awesome, thank you for such a comprehensive suite! :smiley: